### PR TITLE
fix: add support for fiat currency prefixes on NumberDisplay component

### PR DESCRIPTION
**Pull request by ChefBingbong**
---

## PR-Codex overview
This PR focuses on restructuring the `NumberDisplay` components and adding new features like fiat number display formatting.

### Detailed summary
- Deleted `NumberDisplay/index.tsx`
- Added `FiatNumberDisplay.tsx`
- Added `BaseNumberDisplay.tsx`
- Updated `NumberDisplay.tsx`
- Updated `formatNumber` and `formatFiatNumber` functions
- Updated `packages/widgets-internal/index.ts` to include new components

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by ChefBingbong** [comment]
@chefjackson @chef-jojo . could have flag fiatCurrOps wihich if defined will assume the user wants to render a fiat price value by defaulting the valueDisplay to use number.toLocalString with currency option passed by default and symbol from whatever fiatCurrencySymbol is passed. 

decided to keep in the flexibility of allowing the prefix and suffix in this case in the scenario for unique number displays. I like the modularity of the mobile dislay types, but i think something like this (together with a few suggestions ) can work for now

**Review by chefjackson — CHANGES_REQUESTED** [review]
Let's hold until the design has unified the number display format for fiat. And we should build fiat format on top of the general format

File: packages/widgets-internal/components/FiatNumberDisplay/index.tsx
Comment (chefjackson) [inline packages/widgets-internal/components/FiatNumberDisplay/index.tsx:1]: This component should be built on top of the basic NumberDisplay component

**Review by chefjackson — CHANGES_REQUESTED** [review]
Let's build `FiatNumberDisplay` on top of `NumberDisplay`

File: packages/utils/formatNumber.ts [hunk 1]
```diff
@@ -58,10 +71,57 @@ export function formatNumber(
   return `${limitIndicator}${valueToDisplay}`
 }
 
+export function formatFiatNumber({
```
Comment (chefjackson) [inline packages/utils/formatNumber.ts:74]: It's better that we treat `formatFiatNumber` as formatting numbers with a specific rule which suits fiat display. And `formatNumber` is the base function that allow us to specify the rule and `formatFiatNumber` should call `formatNumber` and specify the rule it needs

Comment (ChefBingbong) [reply packages/utils/formatNumber.ts:95]: resolved these. lets chat monday

**Pull request closed without merging**